### ISSUE TITLE
Mill fail to connect to the server with apple silicon M1

### ISSUE THREAD

**Issue by parzival3** [com-lihaoyi/mill#1098]
Step to reproduce the problem
```sh
curl -L https://github.com/lihaoyi/mill/releases/download/0.9.4/0.9.4-13-75bc5a > mill && chmod +x mill
```

and then

```
MILL_VERSION=0.9.4-13-75bc5a ./mill -h
 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   650  100   650    0     0   2241      0 --:--:-- --:--:-- --:--:--  2233
100 58.0M  100 58.0M    0     0  1147k      0  0:00:51  0:00:51 --:--:-- 2148k
main.MillServerMain$.main(MillServerMain.scala:43)
	at mill.main.MillServerMain.main(MillServerMain.scala)
Exception in thread "main" java.lang.Exception: Failed to connect to server
	at mill.main.client.MillClientMain.run(MillClientMain.java:217)
	at mill.main.client.MillClientMain.main0(MillClientMain.java:146)
	at mill.main.client.MillClientMain.main(MillClientMain.java:99)
```

If I use the interactive version everything works 😃 
```
MILL_VERSION=0.9.4-13-75bc5a mill -i -h                                                    
Incomplete argument -h --home <path> is missing a corresponding value
Expected Signature: Mill Build Tool
usage: mill [mill-options] [target [target-options]]
  --no-default-predef  Disable the default predef and run Ammonite with the minimal predef possible
  -s --silent          Make ivy logs go silent instead of printing though failures will still throw
                       exception
  -w --watch           Watch and re-run your scripts when they change
  --bsp                Run a BSP server against the passed scripts
  -c --code <str>      Pass in code to be run immediately in the REPL
  -h --home <path>     The home directory of the REPL; where it looks for config and caches
  -p --predef <path>   Lets you load your predef from a custom location, rather than the "default
                       location in your Ammonite home
[...]

```

**Comment by todor-kolev** [comment]
Hey @parzival3 what is "interactive version"?

**Comment by lefou** [comment]
`mill -i` is the interactive mode. But `mill --no-server` would be sufficient. Both disable the separate server process.

**Comment by BracketMaster** [comment]
This issue still exists and is quite crippling.

**Comment by RickyTino** [comment]
Having the same issue on a VM (x86-64) running Ubuntu 20.04. 
![image](https://user-images.githubusercontent.com/19678521/152720749-4559b4f0-4876-444c-a2de-f93206486a1d.png)

**Comment by lefou** [comment]
I've committed a potential fix for this (#1729), which is currently in release process. Until it's available, you can try the already mentioned `--no-server` Mill option to workaround the issue.

**Comment by lefou** [comment]
Once Mill 0.10.1 is out, no explicit `-i` or `--no-server` should be needed anymore. MIll is able to automatically fall back to in-process mode when the server-process mode isn't working for some reasons. Please report back if you still have issues.
